### PR TITLE
Bug fix in the minor compaction segment size identification.

### PR DESCRIPTION
**Pull request by ravikiran23**
Handling of the locking mechanism for compaction using wrapper thread.
improving the logs for compaction.

### CODE REVIEW

File: integration/spark/src/main/java/org/carbondata/spark/merger/CarbonDataMergerUtil.java [hunk 1]
```diff
@@ -409,13 +407,6 @@ private static boolean isTwoDatesPresentInRequiredRange(Date segDate1, Date segD
       // in case of major compaction the size doesnt matter. all the segments will be merged.
       if (totalLength < (compactionSize * 1024 * 1024)) {
```
Comment (Vimal-Das) [inline integration/spark/src/main/java/org/carbondata/spark/merger/CarbonDataMergerUtil.java]: We should externalize this size limits to carbon.properties

File: integration/spark/src/main/java/org/carbondata/integration/spark/merger/RowResultMerger.java [hunk 2]
```diff
@@ -215,6 +215,7 @@ private CarbonFactDataHandlerModel getCarbonFactDataHandlerModel(CarbonLoadModel
     carbonFactDataHandlerModel.setDatabaseName(schemaName);
     carbonFactDataHandlerModel.setTableName(tableName);
     carbonFactDataHandlerModel.setMeasureCount(segprop.getMeasures().size());
+    carbonFactDataHandlerModel.setCompactionFlow(true);
```
Comment (gvramana) [inline integration/spark/src/main/java/org/carbondata/integration/spark/merger/RowResultMerger.java]: write a comment why this is required

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala [hunk 3]
```diff
@@ -70,7 +70,7 @@ class CarbonMergerRDD[K, V](
       val carbonSparkPartition = theSplit.asInstanceOf[CarbonSparkPartition]
 
       val tempLocationKey: String = carbonLoadModel.getDatabaseName + '_' + carbonLoadModel
-        .getTableName
+        .getTableName + carbonLoadModel.getTaskNo
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonMergerRDD.scala]: You have to use temp location, to pass, cannot set it in CarbonProperties

File: processing/src/main/java/org/carbondata/processing/store/CarbonFactDataHandlerColumnar.java [hunk 4]
```diff
@@ -342,13 +342,25 @@ private void initParameters(CarbonFactDataHandlerModel carbonFactDataHandlerMode
...
-    } catch (NumberFormatException exc) {
-      numberOfCores = Integer.parseInt(CarbonCommonConstants.NUM_CORES_DEFAULT_VAL);
+
+    if (carbonFactDataHandlerModel.isCompactionFlow()) {
+      try {
+        numberOfCores = Integer.parseInt(CarbonProperties.getInstance()
+            .getProperty(CarbonCommonConstants.NUM_CORES_COMPACTING,
+                CarbonCommonConstants.NUM_CORES_DEFAULT_VAL));
+      } catch (NumberFormatException exc) {
+        numberOfCores = Integer.parseInt(CarbonCommonConstants.NUM_CORES_DEFAULT_VAL);
```
Comment (gvramana) [inline processing/src/main/java/org/carbondata/processing/store/CarbonFactDataHandlerColumnar.java]: write logger, falling back to default

File: integration/spark/src/main/java/org/carbondata/integration/spark/merger/RowResultMerger.java (hunk 2 above)
Comment (ravikiran23) [inline integration/spark/src/main/java/org/carbondata/integration/spark/merger/RowResultMerger.java]: added comments.

File: processing/src/main/java/org/carbondata/processing/store/CarbonFactDataHandlerColumnar.java (hunk 4 above)
Comment (ravikiran23) [inline processing/src/main/java/org/carbondata/processing/store/CarbonFactDataHandlerColumnar.java]: writing the logger.

File: integration/spark/src/main/java/org/carbondata/spark/merger/CarbonDataMergerUtil.java (hunk 1 above)
Comment (ravikiran23) [inline integration/spark/src/main/java/org/carbondata/spark/merger/CarbonDataMergerUtil.java]: compactionSize is taken from the carbon properties only.

**Pull request merged**